app/server.go

```diff
@@ -311,8 +312,10 @@ func tryConnect(ip, binding network.Address) error {
 	<-listening
 	conn, err := net.Dial("tcp", ip.NetworkAddress())
 	if err != nil {
-		log.Error("Could not connect itself to public address:", err)
 	} else {
+		log.Panic("Could not connect itself to public address. This is most"+
+			" probably an error in your system-setup. Please make sure this conode "+
+			"can connect to", ip.NetworkAddress())
```

Comment:
That's a strange `if`-branch... `Panic`, then `Info`, then `Close`. Best you do:
```
conn, err :=
log.ErrFatal("Could not...")
log.Info()
conn.Close()
```